src/main/java/com/iotiq/user/security/AuthenticationController.java

```diff
@@ -93,4 +97,18 @@ private <T extends GrantedAuthority> PreAuthenticatedAuthenticationToken createP
         return new PreAuthenticatedAuthenticationToken(authenticationToken.getPrincipal(),
                 authenticationToken.getCredentials(), authorities);
     }
+
+    @GetMapping("/me")
+    public UserDto getCurrentUser(Principal principal) {
+        if (principal == null) throw new InvalidCredentialException();
+        User user = userService.find(principal.getName());
+        return UserDto.of(user);
```

Comment:
Do we need authorities property on profile page?

Reply:
Actually the GET "/me" endpoint is a general purpose endpoint. It typically includes the user's role, resource access permissions and settings, and more. For example, the user re-opened the web page from the browser and with the token from the frontend cache memory, he first sends a request to the /me endpoint and determines how it will look in the UI and which components it will render according to the response of this request.